Abstract logging logic to trait for non-FileTarget logging

timkelty opened the pull request:
This abstracts the non-FileTarget specific logic to a trait, and uses the init methods of `FileTarget` and `LogTrait` to set things dynamically (e.g. from Craft config).

This lets someone make a new logger with all the Craft-added stuff (message formatting, ip logging, etc) as simple as:

```
class MyTarget extends \yii\log\SyslogTarget
{
    use \craft\base\LogTrait;
}
```

brandonkelly commented:
I don’t like the idea of FileTarget configuring itself. 

If you want to take advantage of the changes `craft\log\FileTarget` make over `yii\Log\FileTarget`, then just extend it.

timkelty commented:
> I don’t like the idea of FileTarget configuring itself.

Fair enough, the LogTrait part was really the important part. I reverted the init parts if that is more agreeable.

> If you want to take advantage of the changes craft\log\FileTarget make over yii\Log\FileTarget, then just extend it.

I can't, `craft\log\FileTarget` already extends `yii\Log\FileTarget`, but I don't want FileTarget as my base.
What I'm after is logging using `\yii\log\SyslogTarget` or `codemix\streamlog\Target`, but with the Craft niceties that are currently stuck in `craft\log\FileTarget`, even though they aren't really specific to file logging.

timkelty commented:
@brandonkelly I rebased my branch on latest `develop` and removed the `init` configuring: https://github.com/craftcms/cms/compare/develop...timkelty:log-trait

Apparently PRs don't get updated with new commits once they're closed.

If you still don't like this, let me know if you have other ideas (other than just copying `craft\log\FileTarget` wholesale into my own logger).

brandonkelly commented:
Ah, makes sense now. Just cleaned up some stuff and merged your `log-trait` branch. (fc54c0e04830d2def92d051949b76dc902088d10)

timkelty commented:
Awesome, I appreciate it!
